worker/sshserver/server.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+	"io"
+	"net"
+
+	"github.com/gliderlabs/ssh"
+	"github.com/juju/errors"
+	gossh "golang.org/x/crypto/ssh"
+
+	"github.com/juju/juju/state"
+)
+
+// NewSSHServer returns an embedded SSH server. This server does a few things,
+// reuqests that come in must go through the jump server. The jump server will
+// pipe the connection and pass it into an in-memory instance of another SSH server.
+//
+// This second SSH server (seen within directTCPIPHandlerClosure) will handle the
+// the termination of the SSH connections, note, it is not listening on any ports
+// because we are passing the piped connection to it, essentially allowing the following
+// to work (despite only having one server listening):
+// - `ssh -J controller:2223 ubuntu@app.controller.model`
+//
+// TODO(ale8k): Word this comment better later explaining why the host routing works.
+func NewSSHServer(sp *state.StatePool, jumpHostKey string) (*ssh.Server, error) {
+	jumpHostSigner, err := gossh.ParsePrivateKey([]byte(jumpHostKey))
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	server := &ssh.Server{
+		Addr: ":2223",
+		PublicKeyHandler: func(ctx ssh.Context, key ssh.PublicKey) bool {
+			return true
+		},
+		PasswordHandler: func(ctx ssh.Context, password string) bool {
+			return true
+		},
+		ChannelHandlers: map[string]ssh.ChannelHandler{
+			"direct-tcpip": directTCPIPHandlerClosure(sp),
+		},
+	}
+
+	server.AddHostKey(jumpHostSigner)
+
+	return server, nil
```

Comment:
I wonder if we should wrap the `ssh.Server` in our own server type. Then the `directTCPIPHandlerClosure` can be a server method and not a closure. With that in mind you can then have a tomb on the wrapped server, which each goroutine can be associated with (see other comment). This would mean that the worker would require a catacomb and not a tomb.

Reply:
I did this originally, but it ended up a bit weird:
- I created the wrapper
- The wrapper had the direct handler and dependencies
- I created the ssh server
- The ssh server configured the channel handler as a method of the wrapper
- The wrapper never necessarily needed to embed the server

So it felt a bit wrong really? Where as the closure just given the necessary dependencies and is fairly obvious what it's doing...

In resp to the matrix message, we can hop on tomorrow and have a look together again - I think that'd be good. Plus I'm not familiar with catacomb so that'd be good.